apps/prairielearn/src/sync/sharing.ts

```diff
@@ -185,3 +185,39 @@ export async function checkInvalidSharingSetRemovals(
 
   return existInvalidSharingSetRemovals;
 }
+
+export function checkInvalidSharedAssessments(
+  courseData: CourseData,
+  logger: ServerJobLogger,
+): boolean {
+  const invalidSharedAssessments: string[] = [];
+  for (const courseInstanceKey in courseData.courseInstances) {
+    const courseInstance = courseData.courseInstances[courseInstanceKey];
+    for (const tid in courseInstance.assessments) {
+      const assessment = courseInstance.assessments[tid];
+      if (!assessment?.data?.shareSourcePublicly) {
+        continue;
+      }
+      for (const zone of assessment?.data?.zones ?? []) {
+        for (const question of zone.questions ?? []) {
+          if (!question.id) {
+            continue;
+          }
+          const infoJson = courseData.questions[question.id];
+          if (!infoJson?.data?.sharedPublicly) {
+            invalidSharedAssessments.push(tid);
+            continue;
```

Comment:
I just realized that the continue here will only jump out of the loop over question, and the loop over zones will keep going. That means that a `tid` could still end up in the array multiple times. The simplest solution is probably just to change `invalidSharedAssessments` to be a `Set<string>` instead of `string[]`, then it will just de-dupe itself and we can get rid of this continue statement.

Reply:
I made it so the 'shareSourcePublicly' property is removed in the AssessmentCopyEditor as requested above

Reply:
Changed invalidSharedAssessments to be a Set

Reply:
Changed it to use a Set<string> and updated all calls on it to work with the 'Set' object